pkg/handlers/handlers.go

```diff
@@ -255,24 +231,111 @@ func (h *Handlers) processResults(uid uuid.UUID, runResultsChan chan runs.Profil
 	} else {
 		// no errors : simply log the results
 		hlog.Info(logMessage.String())
-		_, err := writeRunToLogFile(arun, h.StorageFolder)
-		if err != nil {
+		if err := writeRunToFile(arun, h.runLogOutputFilePath(arun)); err != nil {
 			hlog.Fatal(err)
 		}
 	}
 }
 
-func writeRunToLogFile(arun runs.Run, storageFolder string) (string, error) {
+// outputFilePath returns the full file path from the storage folder.
+func (h *Handlers) outputFilePath(prefix, id, ext string) string {
+	if prefix != "" {
+		prefix = prefix + "-"
+	}
+	return filepath.Join(h.StorageFolder, prefix+id+"."+ext)
+}
 
-	fileName := filepath.Join(storageFolder, arun.ID.String()+"."+string(logFile))
+// crioPprofOutputFilePath returns the full file path for CRIO pprof output.
+func (h *Handlers) crioPprofOutputFilePath(id string) string {
+	return h.outputFilePath(crioFilePrefix, id, pprofFileExt)
+}
 
-	bytes, err := json.Marshal(arun)
+// kubeletPprofOutputFilePath returns the full file path for Kubelet pprof output.
+func (h *Handlers) kubeletPprofOutputFilePath(id string) string {
+	return h.outputFilePath(kubeletFilePrefix, id, pprofFileExt)
+}
+
+// runLogOutputFilePath returns the full file path for CRIO pprof output.
```

Comment:
```suggestion
// runLogOutputFilePath returns the full file path for pprof output.
```
Seems to be a generic func.